common/static/js/src/accessibility_tools.js

```diff
@@ -33,7 +33,111 @@ NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVE
+    // Sets appropriate elements to tab indexable and properly sets aria_hidden on content outside of modal
+    // "focusableElementsString" is a string that indicates all elements that should be focusable
+    // "closeButtonId" is the selector for the button that closes out the modal.
+    // "mainPageId" is the selector for the main part of the page
+    // "modalId" is the selector for the modal being managed
+    // Returns a list of focusableItems
+    var focusableItems;
+
+    $(mainPageId).attr('aria-hidden', 'true');
+    $(modalId).attr('aria-hidden', 'false');
+
+    focusableItems = $(modalId).find('*')
+        .filter(focusableElementsFilterString)
+        .filter(':visible');
+
+    focusableItems.attr('tabindex', '2');
+    $(closeButtonId).attr('tabindex', '1').focus();
+
+    return focusableItems;
+};
+
+var trapTabFocus = function(focusableItems, closeButtonId) {
+    // Determines last element in modal and traps focus by causing tab
+    // to focus on the first modal element (close button)
+    // "focusableItems" all elements in the modal that are focusable
+    // "closeButtonId" is the selector for the button that closes out the modal.
+    // returns the last focusable element in the modal.
+    var $last;
+    if (focusableItems.length !== 0) {
+        $last = focusableItems.last();
+    } else {
+        $last = $(closeButtonId);
+    }
+
+    // tab on last element in modal returns to the first one
+    $last.on('keydown', function(e) {
+        var keyCode = e.keyCode || e.which;
+        // 9 is the js keycode for tab
+        if (!e.shiftKey && keyCode === 9) {
+            e.preventDefault();
+            $(closeButtonId).focus();
+        }
+    });
+
+    return $last;
+};
+
+var trapShiftTabFocus = function($last, closeButtonId) {
+    $(closeButtonId).on('keydown', function(e) {
+        var keyCode = e.keyCode || e.which;
+        // 9 is the js keycode for tab
+        if (e.shiftKey && keyCode === 9) {
+            e.preventDefault();
+            $last.focus();
+        }
+    });
+};
+
+var bindReturnFocusListener = function($previouslyFocusedElement, closeButtonId, modalId, mainPageId) {
+    // Ensures that on modal close, focus is returned to the element
+    // that had focus before the modal was opened.
+    $('#lean_overlay, ' + closeButtonId).click(function() {
+        $(mainPageId).attr('aria-hidden', 'false');
+        $(modalId).attr('aria-hidden', 'true');
+        $previouslyFocusedElement.focus();
+    });
+};
+
+var bindEscapeKeyListener = function(modalId, closeButtonId) {
+    $(modalId).on('keydown', function(e) {
+        var keyCode = e.keyCode || e.which;
+        // 27 is the javascript keycode for the ESC key
+        if (keyCode === 27) {
+            e.preventDefault();
+            $(closeButtonId).click();
+        }
+    });
+};
+
+var trapFocusForAccessibleModal = function(
+    $previouslyFocusedElement,
+    focusableElementsMatchString,
```

Comment:
Are `focusableElementsMatchString`, `focusableElementsString`, and `focusableElementsFilterString` all distinct? If not, it might be useful to pick one name for all of them, so it's clear to future devs that they are equivalent.

Reply:
They are pretty much the same, I was trying to differentiate between the actual value at the top of the file, and the value as a required parameter for the functions, (also I get linting errors around having function parameters with the same name as an already declared variable, thanks Javascript). I'll trying leaving it the same for all the parameters but different for the value itself.

Reply:
I had considered there might be annoying javescript reasoning for the different names, just wanted to call it out in case it was just coincidence that the names were different